java/feathub-connectors/flink-connectors/flink-connector-prometheus/src/main/java/com/alibaba/feathub/flink/connectors/PrometheusSinkFunction.java

```diff
@@ -0,0 +1,181 @@
+
+/**
+ * A {@link org.apache.flink.streaming.api.functions.sink.SinkFunction} that writes to a Prometheus
+ * PushGateway.
+ *
+ * <p>Note that the SinkFunction only support writing numeric columns.
+ */
+public class PrometheusSinkFunction extends RichSinkFunction<RowData> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PrometheusSinkFunction.class);
+
+    private final ReadableConfig config;
+    private final List<String> columnNames;
+    private final List<String> columnComments;
+    private CollectorRegistry registry;
+    private PushGateway pushGateway;
+    private String jobName;
+    private Map<String, String> extraLabels;
+    private Gauge[] metrics;
+    private final List<DataType> fieldTypes;
+
+    public PrometheusSinkFunction(ReadableConfig config, ResolvedSchema schema) {
+        this.config = config;
+        this.fieldTypes = schema.getColumnDataTypes();
+        this.columnNames =
+                schema.getColumns().stream().map(Column::getName).collect(Collectors.toList());
+        this.columnComments =
+                schema.getColumns().stream()
+                        .map(c -> c.getComment().orElse(c.getName()))
+                        .collect(Collectors.toList());
+    }
+
+    @Override
+    public void open(Configuration parameters) throws Exception {
+        super.open(parameters);
+
+        registry = new CollectorRegistry();
+        pushGateway = new PushGateway(config.get(HOST_URL));
+        jobName = config.get(JOB_NAME);
+        extraLabels = parseGroupingKey(config.get(EXTRA_LABELS));
+
+        metrics = new Gauge[columnNames.size()];
+        for (int i = 0; i < columnNames.size(); i++) {
+            final LogicalType logicalType = fieldTypes.get(i).getLogicalType();
+            if (!logicalType.is(LogicalTypeFamily.NUMERIC)) {
+                throw new IOException("PrometheusSink only support numeric data types.");
+            }
+
+            metrics[i] =
+                    Gauge.build()
+                            .name(columnNames.get(i))
+                            .help(columnComments.get(i))
+                            .register(registry);
+        }
+    }
+
+    @Override
+    public void invoke(RowData row, Context context) throws Exception {
+        for (int i = 0; i < metrics.length; i++) {
+            final Gauge metric = metrics[i];
+            final LogicalType type = fieldTypes.get(i).getLogicalType();
+            switch (type.getTypeRoot()) {
+                case DECIMAL:
+                    final int precision = ((DecimalType) type).getPrecision();
+                    final int scale = ((DecimalType) type).getScale();
+                    final DecimalData decimal = row.getDecimal(i, precision, scale);
+                    metric.set(decimal.toBigDecimal().doubleValue());
+                    break;
+                case TINYINT:
+                    metric.set(row.getByte(i));
+                    break;
+                case SMALLINT:
```

Comment:
I think supporting the additional type of decimal, tinyint and smallint doesn't add much complexity to the code base and it can be more future-proof, i.e. Feathub may support those types in the future. What do you think?

Reply:
I'm OK with it given that we have added test cases that cover these data types in this PR.